src/util.ts

```diff
@@ -167,24 +169,23 @@ const DAY_STR_LENGTH = 2;
  * @param {number} year - A calendar year.
  * @param {number} [month] - A calendar month.
  * @param {number} [day] - A calendar day of month.
- * @returns {string} The provided date formatted as an ISO 8601-2004 year or calendar
- *                   date.
+ * @returns {string} The provided date formatted as an ISO 8601-2004 year or calendar date.
  */
-export function formatDate(year, month, day) {
-	if ((!year || isNaN(parseInt(year, 10))) && year !== 0) {
+export function formatDate(year: number, month?: number, day?: number): string {
+	if ((!year || isNaN(parseInt(year as any, 10))) && year !== 0) {
```

Comment:
We can't get away with dropping `!year` and the like as they also catch `null` and `undefined` values (confirmed 14 failing tests when I tried that). So I've only used `_.toNumber` in 5a519ec9e7e166e97833e9ed987c646da5b73139 to avoid using `as any`.